src/Universe.cpp

```diff
@@ -1112,6 +1123,34 @@ void Lattice::removeUniverse(Universe* universe) {
 }
 
 
+/**
+ * @brief Subdivides all of the Material-filled Cells within this Lattice 
+ *        into rings and angular sectors aligned with the z-axis.
+ * @param max_radius the maximum allowable radius used in the subdivisions
+ */
+void Lattice::subdivideCells(double max_radius) {
+
+  log_printf(DEBUG, "Subdividing Cells for Lattice ID=%d "
+             "with max radius %f", _id, max_radius);
+
+  std::map<int, Universe*>::iterator iter;
+  std::map<int, Universe*> universes = getUniqueUniverses();
+
+  /* Compute minimum radius needed to surround a lattice cell */
+  /* This is used as the maximum radius for all ringified Cells */
+  double radius = sqrt(_width_x * _width_x / 2. + _width_y * _width_y / 2.);
+
+  /* If the lattice pitch is smaller than max_radius parameter, over-ride it */
+  if (radius < max_radius)
+    max_radius = radius;
+
+  /* Subdivide all Cells */
+  while (iter != universes.end()) {
```

Comment:
Great catch, it's nice how the discretization algorithms have evolved and simplified to this point.